From 8b77b0b112aa881102a2a497c80e777d96c4ea49 Mon Sep 17 00:00:00 2001
From: James Simmons <uja.ornl@yahoo.com>
Date: Wed, 02 Jan 2019 11:25:58 -0500
Subject: [PATCH] LU-11742 test: have libtool execute the test binaries

With the move to libtools the ability to run all the lustre
utilities form the source tree was lost. To work around this
the libtool -no-install flag was used to prevent the creation
of the libtool wrappers. While this worked to restore the
source tree sand box development new package breakage is showing.
This is due to the rpath being hard coded into the utilies when
-no-install is used and some platforms disable fixed rpaths.

A very similar problem exist for people who want to use gdb to
debug their projects application. gdb does not work on libtool
wrappers as well so the recommended approach to this type of
problem is to use the libtool --mode=e command. This command
allows the exectuion of an external non project binary, like
gdb, with the projects real binary application. Apply this
approach to the lustere test suite so commands like kill can
be used to shutdown lustre utilies that are not installed into
the testing environment.

Change-Id: I74112f7250f1c43313d868c0edc7c8815d373002
Signed-off-by: James Simmons <uja.ornl@yahoo.com>
---

diff --git a/lustre/tests/Makefile.am b/lustre/tests/Makefile.am
index 1e9f6ec..c77958c 100644
--- a/lustre/tests/Makefile.am
+++ b/lustre/tests/Makefile.am
@@ -1,7 +1,6 @@
 # Lustre test Makefile
 AM_CFLAGS := -fPIC -D_GNU_SOURCE \
 	     -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64
-AM_LDFLAGS := -no-install
 
 DIST_SUBDIRS = mpi
 
diff --git a/lustre/tests/sanity-hsm.sh b/lustre/tests/sanity-hsm.sh
index 2035603..3a91170 100755
--- a/lustre/tests/sanity-hsm.sh
+++ b/lustre/tests/sanity-hsm.sh
@@ -150,14 +150,14 @@
 
 search_copytools() {
 	local hosts=${1:-$(facet_active_host $SINGLEAGT)}
-	do_nodesv $hosts "pgrep -x $HSMTOOL_BASE"
+	do_nodesv $hosts "libtool --mode=e pgrep -x $HSMTOOL_BASE"
 }
 
 kill_copytools() {
 	local hosts=${1:-$(facet_active_host $SINGLEAGT)}
 
 	echo "Killing existing copytools on $hosts"
-	do_nodesv $hosts "killall -q $HSMTOOL_BASE" || true
+	do_nodesv $hosts "libtool --mode=e killall -q $HSMTOOL_BASE" || true
 	copytool_continue "$hosts"
 }
 
@@ -288,7 +288,7 @@
 	cmd+=" \"$mountpoint\""
 
 	echo "Starting copytool $facet on $(facet_host $facet)"
-	stack_trap "do_facet $facet pkill -x '$HSMTOOL_BASE' || true" EXIT
+	stack_trap "do_facet $facet libtool --mode=e pkill -x '$HSMTOOL_BASE' || true" EXIT
 	do_facet $facet "$cmd < /dev/null > \"$(copytool_logfile $facet)\" 2>&1"
 }
 
@@ -400,15 +400,15 @@
 	local agents=${1:-$(facet_active_host $SINGLEAGT)}
 
 	stack_trap \
-		"do_nodesv $agents pkill -CONT -x '$HSMTOOL_BASE' || true" EXIT
-	do_nodesv $agents "pkill -STOP -x $HSMTOOL_BASE" || return 0
+		"do_nodesv $agents libtool --mode=e pkill -CONT -x '$HSMTOOL_BASE' || true" EXIT
+	do_nodesv $agents "libtool --mode=e pkill -STOP -x $HSMTOOL_BASE" || return 0
 	echo "Copytool is suspended on $agents"
 }
 
 copytool_continue() {
 	local agents=${1:-$(facet_active_host $SINGLEAGT)}
 
-	do_nodesv $agents "pkill -CONT -x $HSMTOOL_BASE" || return 0
+	do_nodesv $agents "libtool --mode=e pkill -CONT -x $HSMTOOL_BASE" || return 0
 	echo "Copytool is continued on $agents"
 }
 
@@ -853,7 +853,7 @@
 
 	# Lustre mount-point is mandatory and last parameter on
 	# copytool cmd-line.
-	local mntpnt=$(do_rpc_nodes $agent ps -C $HSMTOOL_BASE -o args= |
+	local mntpnt=$(do_rpc_nodes $agent libtool --mode=e ps -C $HSMTOOL_BASE -o args= |
 		       awk '{print $NF}')
 	[ -n "$mntpnt" ] || error "Found no Agent or with no mount-point "\
 				  "parameter"
diff --git a/lustre/utils/Makefile.am b/lustre/utils/Makefile.am
index c7dd164..d30d6ab 100644
--- a/lustre/utils/Makefile.am
+++ b/lustre/utils/Makefile.am
@@ -2,7 +2,6 @@
 
 AM_CFLAGS  := -fPIC -D_GNU_SOURCE \
 	      -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DLUSTRE_UTILS=1
-AM_LDFLAGS := -no-install
 
 if TESTS
 EXTRA_PROGRAMS = wirecheck
diff --git a/lustre/utils/gss/Makefile.am b/lustre/utils/gss/Makefile.am
index 437a2e5..dff542d 100644
--- a/lustre/utils/gss/Makefile.am
+++ b/lustre/utils/gss/Makefile.am
@@ -3,7 +3,6 @@
 AM_CFLAGS := -fPIC \
 	     -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DLUSTRE_UTILS=1 \
 	     -D_GNU_SOURCE
-AM_LDFLAGS := -no-install
 
 sbin_PROGRAMS = lsvcgssd l_idmap
 
